ansible/roles/gpu/tasks/nvidia-gpu-RHEL8.yaml

```diff
@@ -27,18 +33,28 @@
     gpgkey: "{{ nvidia_repo_gpgkey }}"
     gpgcheck: true
 
-- name: Install cuda drivers and tools
-  dnf:
-    enablerepo:
-      - "codeready-builder-for-rhel-8-{{ ansible_architecture }}-rpms"
-      - "rhel-8-for-{{ ansible_architecture }}-baseos-rpms"
-      - "rhel-8-for-{{ ansible_architecture }}-appstream-rpms"
-    name:
-      - "{{ nvidia_cuda_package }}"
-    state: present
+- name: create temporary run file directory
+  ansible.builtin.tempfile:
+    state: directory
+    suffix: build
+  register: runfile_dir
+
+# It appears impossible to pin a specific CUDA version on RHEL8 unless
+# one uses the runfile installer.
+# Installing `cuda-XX-X` will result in sliding CUDA versions.
+# TODO(tillt): Find a way that spares us the runfile.
+- name: download Nvidia driver installer
+  get_url:
+    url: "{{ nvidia_runfile_installer_url }}"
+    dest: "{{ runfile_dir.path }}/{{ nvidia_runfile_installer }}"
+    mode: 0755
+  retries: 3
+  delay: 3
+
+- name: run Nvidia driver installer
+  command: "{{ runfile_dir.path }}/{{ nvidia_runfile_installer }} --silent"
 
```

Comment:
Do we want this to only run on 8.4? If so, can you put the appropriate `when`?

Reply:
No, good as is - we want the driver installed like this for all RHEL deployments.